src/lib/fcitx/inputcontext_p.h

```diff
@@ -132,6 +133,14 @@ class InputContextPrivate : public QPtrHolder<InputContext> {
 
     InputContextProperty *property(int slot) { return properties_[slot].get(); }
 
+    bool isFunctionModeSupported(FunctionMode functionMode) {
+        FCITX_Q();
+
+        auto *instance = manager_.instance();
+        auto *engine = instance->inputMethodEngine(q);
```

Comment:
engine can be null, even instance can be null.

Reply:
OK. I will fix it.